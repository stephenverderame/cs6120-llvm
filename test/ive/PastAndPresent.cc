// NOLINTBEGIN
#include <cstdio>
int main()
{
    int i = 0;
    while (i < 100) {
        int k = i * 10 + 5;
        printf("%d\n", k);
        ++i;
        int m = i * 20 + 5;
        printf("%d\n", m);
    }
}
// NOLINTEND