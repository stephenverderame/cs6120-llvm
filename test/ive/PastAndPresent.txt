5
25
15
45
25
65
35
85
45
105
55
125
65
145
75
165
85
185
95
205
105
225
115
245
125
265
135
285
145
305
155
325
165
345
175
365
185
385
195
405
205
425
215
445
225
465
235
485
245
505
255
525
265
545
275
565
285
585
295
605
305
625
315
645
325
665
335
685
345
705
355
725
365
745
375
765
385
785
395
805
405
825
415
845
425
865
435
885
445
905
455
925
465
945
475
965
485
985
495
1005
505
1025
515
1045
525
1065
535
1085
545
1105
555
1125
565
1145
575
1165
585
1185
595
1205
605
1225
615
1245
625
1265
635
1285
645
1305
655
1325
665
1345
675
1365
685
1385
695
1405
705
1425
715
1445
725
1465
735
1485
745
1505
755
1525
765
1545
775
1565
785
1585
795
1605
805
1625
815
1645
825
1665
835
1685
845
1705
855
1725
865
1745
875
1765
885
1785
895
1805
905
1825
915
1845
925
1865
935
1885
945
1905
955
1925
965
1945
975
1965
985
1985
995
2005
