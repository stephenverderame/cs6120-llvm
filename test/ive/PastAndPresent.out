Loop: 
1:                                                ; preds = %3, %0
  %.0 = phi i32 [ 0, %0 ], [ %7, %3 ]
  %2 = icmp slt i32 %.0, 100
  br i1 %2, label %3, label %11

Basic IV:   %.0 = phi i32 [ 0, %0 ], [ %7, %3 ]
Derived IV: %4 = %.0 * 10
Derived IV: %5 = %.0 * 10 + 5
Basic IV:   %7 = add nsw i32 %.0, 1
Derived IV: %8 = %7 * 20
Derived IV: %9 = %7 * 20 + 5
